### ISSUE TITLE
Warn when negative numbers are raised to the 0th power

### ISSUE THREAD

**Issue by GriffinLeclerc** [rust-lang/rust-clippy#10833]
### What it does

When negative numbers are raised to the 0th power in Rust, the exponent is evaluated before the sign is applied.
For example:
-1_i32.pow(0) evaluates to -1.  
Humans most likely expect this to evaluate this to 1, as any non-zero number raised to the 0th power equals 1. 

Here is a playground permalink showcasing this:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=115862da3c4e35ef87ab3b20bee1687b

This issue is overcome by parenthesizing the integer: (-1_i32).pow(0) but could instead be linted to avoid unnecessary confusion.  

### Lint Name

negative_to_0th_power

### Advantage

- Removes the possibility of human error when dealing with 0 exponents.

### Drawbacks

- Potential for overzealous application, where if implemented improperly this lint may provide false positives.

### Example

```rust
-1_i32.pow(0)
```

Could be written as:

```rust
(-1_i32).pow(0)
```

**Comment by blyxyas** [comment]
We already have the [`clippy::precedence`](https://rust-lang.github.io/rust-clippy/master/index.html#/precedence). I think that lint already does what you're thinking about.

```
error: unary minus has lower precedence than method call
  --> $DIR/negative_to_0th_power.rs:6:11
   |
LL |     let _ = -1.pow(0);
   |              ^^^^^^^^^ help: consider adding parentheses to clarify your intent: `-(1.pow(0))`
   |
   = note: `-D clippy::precedence` implied by `-D warnings`
```

Maybe changing the suggestion would fix this issue?

**Comment by Alexendoo** [comment]
Oh, so we do. Changing the suggestion to `(-1).pow(0)` would be incorrect, the added parens make it clear though so I'd say there's nothing to be done here
